docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,247 @@
+  - MJML has by far the highest weekly downloads and github stars in comparision to its alternatives like Foundation for Emails
+- Cons
+  - Although MJML is a responsive library, it controls that responsiveness automatically needing very less to no developer intervention
+  - Adds a new dependency to the stack
+  - Creates a slight learning curve
+
+## Pros and Cons of the Templating Options
+
+`fxa-auth-server` currently makes use of Mustache to develop email templates. Mustache is a logicless HTML templating framework and has been proven to be super helpful in developing mail components so far, but due to its logicless nature it becomes tedious to include conditional statements in the mail components. The following section will detail upon the various pros and cons of potential libraries/ frameworks that could be used to replace Mustache as a templating solution.
+
+### Option A - Continue to use Mustache
+
+- Description
+  - Keep the email templates as is in Mustache
+- Pros
+  - The effort to refactor all the templates would be eliminated for now
+- Cons
+  - The logicless nature of Mustache would continue to bother us in the future
+  - If we continue to use Mustache for our templates then our front-end frameworks in FxA would continue to be different. We currently have Backbone in the content-server, Mustache in the auth-server, and React everywhere else.
+
+### Option B - Use React server-side to generate static pages
+
+- Description
+  - Template the auth server emails using [React](https://reactjs.org/) server-side and convert them over to [static HTML pages](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup). Other pros and cons of React can be found [in this ADR](https://github.com/mozilla/fxa/blob/main/docs/adr/0010-transition-fxa-from-backbone-to-react.md#option-c---move-to-isomorphic-react), though the type of React we're describing here would be stateless.
+- Pros
+  - The logicless nature of the templates will be eliminated once they're refactored in React
+  - We already use React in `fxa-settings`, `fxa-admin-panel`, `fxa-payments-server`, and `fxa-react`. There’s potential to cross-share components with the new email templates.
+  - React offers us many integration options. E.g there is one with [MJML](https://github.com/wix-incubator/mjml-react)
+  - React has already been audited by our security team.
+- Cons
+  - It could be a overkill to use React, since the library comes with far too many features which may not be needed for templating
+  - The templates in the `fxa-auth-server` do not contain heavy business logic so building them with a frontend-heavy framework like React might complicate things
+  - Since we will be converting React components into static HTML pages, another layer of complexity as well as new dependencies would be added to the project
+
```

Comment:
Oh, another con is I'm pretty sure we're going to have to use `dangerouslySetInnerHTML` tag which may have a bad code smell for some folks when they first see it, but we'll be safe using it since we're generating static pages from the server and won't have a risk of XSS attacks.

Reply:
Actually, I don't think we'll need to use that so I'll mark this as resolved. I was thinking of some other project I worked on with stateless React where sometimes the strings we used contained HTML tags but we shouldn't need to do that.